modal/_functions.py

```diff
@@ -1865,6 +1902,47 @@ async def get(self, timeout: Optional[float] = None, *, index: int = 0) -> Retur
         """
         return await self._invocation().poll_function(timeout=timeout, index=index)
 
+    @live_method_gen
+    async def iter(self, *args) -> AsyncIterator[Tuple[int, ReturnType]]:
+        """Iterate in-order over (index, result) pairs of the function call.
```

Comment:
I might expect a method that returns an (index, result) pair to be called something like "enumerate".

OTOH would some name that includes "get" help clarify what this is? IDK if `FunctionCall.iter_get()` is sort of akward though.